libibumad/umad.h

```diff
@@ -62,7 +62,7 @@ union umad_gid {
 	} global;
 } __attribute__((aligned(4))) __attribute__((packed));
 
-#define UMAD_MAX_DEVICES 32
```

Comment:
While you are here please remove this internal constant from the public header.

It would be better to make a better interface than umad_get_cas_names() so we don't have such static limits in the first place.

Reply:
UMAD_MAX_DEVICES is used in other utilities/applications.
it would be problematic to move it from the public header.

Reply:
Mark it an attribute deprecated enum then.

Reply:
Did you mean to deprecated macro, not enum, right ?

Reply:
You can't use attribute deprecated on a #define

Reply:
Should it be done in a separate pull request?

Reply:
A separate patch in this pull request is fine.

Reply:
Do you have an example code how to mark macros as deprecated?
Because there is no a standard/clean way to do it.

Reply:
```
enum { UMAD_MAX_DEVICES __attribute__((deprecated)) = 32};
```

Reply:
Thanks, your suggestion is good but it is supported from gcc-6 and above.
Attached link:
https://gcc.gnu.org/gcc-6/changes.html

Reply:
Wrap it in a if gcc version test?

Reply:
Or I can do something like this:
enum __attribute__ ((deprecated)) {
        UMAD_MAX_DEVICES = 128
};

or this:
enum {
        UMAD_MAX_DEVICES = 128
} __attribute__ ((deprecated));